### PR TITLE
Issue-15: Add support for Maybe#filter(predicate)

### LINKED ISSUE #15: Add support for Maybe#filter(predicate)
* Passing the `predicate` returns the `Just` of the value.
* Failing the `predicate` or being a `Nothing` returns `Nothing`.

### PR DESCRIPTION
**Pull request by jlmorgan**
- [x] Squash commits before merging.

closes #15
SemVer:Minor
enhancement

### CODE REVIEW

File: src/data/Maybe/Maybe.js [hunk 1]
```diff
@@ -19,6 +20,34 @@ class Maybe {
     return value instanceof Maybe;
   }
 
+  /**
```
Comment (doomchild) [inline src/data/Maybe/Maybe.js:23]: Did you just put these comments in as placeholders for later code?

**Review by doomchild — COMMENTED** [review]
There are some unnecessary comments in `Maybe.js`.  Are they just placeholders?

File: src/data/Maybe/Maybe.js (hunk 1 above)
Comment (jlmorgan) [reply src/data/Maybe/Maybe.js:23]: They are in there for suppose auto-detection by editors that understand JSDoc.

Comment (doomchild) [reply src/data/Maybe/Maybe.js:23]: So you're adding documentation for functions that don't exist so that editors will auto-detect it?

Comment (jlmorgan) [reply src/data/Maybe/Maybe.js:23]: At least, they are suppose to. = p

Comment (doomchild) [reply src/data/Maybe/Maybe.js:23]: I'm confused.  The editors will auto-detect documentation for functions that don't exist?  I don't understand your endgame here.

Comment (jlmorgan) [reply src/data/Maybe/Maybe.js:23]: There's a competing element here. If I define functions for inheritance with no implementation, but clearly defined JSDoc, JSDoc will complain that they don't match. If I don't do anything in the methods, ESLint will complain. So, by defining the methods in JSDoc, the intent is that the extending classes will inherit that same "interface". This all stems from the lack of `Interface` as a thing in JS.

**Pull request merged**